server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/model/Keywords.scala

```diff
@@ -106,21 +103,11 @@ case class Keywords(keywords: Set[Keyword]) {
     flags
   }
 
-  def asMap: ImmutableMap[String, Boolean] = keywords.asJava.stream()
-    .collect(Guavate.toImmutableMap((keyword: Keyword) => keyword.getFlagName(), (_: Keyword) => Keyword.FLAG_VALUE))
+  def asMap: Map[String, Boolean] = keywords zip Iterable(Keyword.FLAG_VALUE) toMap
```

Comment:
You should check but i think that it will not work, your map will contain only one element.
As with zip the length of the resulting collection is the minimum length of the two collections zipped.

Reply:
`keywords.map((_, Keyword.FLAG_VALUE)).toMap` should do the trick